src/schema/models/schemaManager.ts

```diff
@@ -81,42 +88,49 @@ export class SchemaManager {
       if (domainFields.has(key.toUpperCase()) && value !== null) {
         redisKeysArr.push(`${key.toUpperCase()}:${value.toString()}`);
       }
-      return { ...acc, [`${name}_${key}`]: value };
+      return { ...acc, [this.concatenateKeysPrefix(schema.addSchemaPrefix, name, key)]: value };
     }, {});
 
     let domainFieldsKeys = {};
     let explodeFieldsKeys = {};
 
     if (redisKeysArr.length > 0) {
-      domainFieldsKeys = await this.getDomainFields(redisKeysArr, name);
+      domainFieldsKeys = await this.getDomainFields(redisKeysArr, name, schema.addSchemaPrefix);
     }
     if (explodeKeysArr.length > 0) {
-      explodeFieldsKeys = await this.getExplodeFields(explodeKeysArr, name);
+      explodeFieldsKeys = await this.getExplodeFields(explodeKeysArr, name, schema.addSchemaPrefix);
     }
     finalTagsObj = { ...finalTagsObj, ...domainFieldsKeys, ...explodeFieldsKeys };
     return finalTagsObj;
   }
 
-  public getDomainFields = async (redisKeysArr: string[], name: string): Promise<Tags> => {
+  private readonly concatenateKeysPrefix = (addSchemaPrefix: boolean, prefix: string, ...keys: string[]): string => {
+    return addSchemaPrefix ? `${[prefix, ...keys].join(SEPARATOR)}` : keys[0];
+  };
+
+  private readonly getDomainFields = async (redisKeysArr: string[], name: string, addSchemaPrefix: boolean): Promise<Tags> => {
     let domainFieldsTags: Tags = {};
     const redisRes = await this.domainFieldsRepo.getFields(redisKeysArr);
 
     //for each domain field create new domain field tag with the correct value
     redisRes.forEach((key, index) => {
-      domainFieldsTags = { ...domainFieldsTags, [`${name}_${redisKeysArr[index].split(':')[0]}_DOMAIN`]: key };
+      domainFieldsTags = {
+        ...domainFieldsTags,
+        [`${this.concatenateKeysPrefix(addSchemaPrefix, name, redisKeysArr[index].split(':')[0], 'DOMAIN')}`]: key,
```

Comment:
There is no need to template the string.

Reply:
done